src/test/java/uk/ac/ebi/subs/biostudies/validator/ProjectValidatorTest.java

```diff
@@ -0,0 +1,119 @@
+
+    private ProjectValidator validator;
+    private String gibberishText;
+
+    @Before
+    public void buildUp() {
+        validator = new ProjectValidator();
+        gibberishText = makeText(100);
+    }
+
+    @Test
+    public void testHappyTrail() {
+        Project p = project();
+        p.setReleaseDate(LocalDate.MIN);
+        p.setTitle(gibberishText);
+        p.setDescription(gibberishText);
+
+
+        ValidationMessageEnvelope<Project> envelope = new ValidationMessageEnvelope<>();
+        envelope.setEntityToValidate(p);
+
+        SingleValidationResultsEnvelope validationResultsEnvelope = validator.validateProject(envelope);
+
+        Assert.assertEquals(1, validationResultsEnvelope.getSingleValidationResults().size());
+
+        Assert.assertTrue(validationResultsEnvelope.getSingleValidationResults().get(0).getValidationStatus().equals(SingleValidationResultStatus.Pass));
+
+        Assert.assertEquals(null, validationResultsEnvelope.getSingleValidationResults().get(0).getMessage());
+    }
+
+    @Test
+    public void testHorribleTrail() {
+        Project p = project();
+        p.setReleaseDate(null);
+        p.setTitle("short");
+        p.setDescription("short");
+
+        ValidationMessageEnvelope<Project> envelope = new ValidationMessageEnvelope<>();
+        envelope.setEntityToValidate(p);
+
+        SingleValidationResultsEnvelope validationResultsEnvelope = validator.validateProject(envelope);
+
+        Assert.assertEquals(3, validationResultsEnvelope.getSingleValidationResults().size());
+
+        validationResultsEnvelope.getSingleValidationResults().stream()
+                .forEach(result -> {
+
+                    Assert.assertEquals(SingleValidationResultStatus.Error, result.getValidationStatus());
+                    Assert.assertTrue(result.getMessage() != null);
+
+                });
+    }
+
+    @Test
+    public void testEmptyTrail() {
+        Project p = project();
+        p.setReleaseDate(null);
+        p.setTitle(null);
+        p.setDescription(null);
+
+        ValidationMessageEnvelope<Project> envelope = new ValidationMessageEnvelope<>();
+        envelope.setEntityToValidate(p);
+
+        SingleValidationResultsEnvelope validationResultsEnvelope = validator.validateProject(envelope);
+
+        Assert.assertEquals(3, validationResultsEnvelope.getSingleValidationResults().size());
+
+        validationResultsEnvelope.getSingleValidationResults().stream()
+                .forEach(result -> {
+
+                    Assert.assertEquals(SingleValidationResultStatus.Error, result.getValidationStatus());
+                    Assert.assertTrue(result.getMessage() != null);
+
+                });
+    }
+
+    static Project project() {
+        Project p = new Project();
+        p.setTeam(Team.build("test"));
+        p.setAlias("alias");
+
+        return p;
+    }
+
+    static ValidationMessageEnvelope<Project> generateValidationMessageEnvelope(Project project) {
```

Comment:
Are we using this method somewhere? I could not find it.

Reply:
Indeed we are not